src/main/scala/com/databricks/labs/overwatch/utils/Config.scala

```diff
@@ -381,6 +382,43 @@ class Config() {
     }
   }
 
+  private def validateApiEnvConfig(apiEnvConfig: Option[ApiEnvConfig]) = {
+    if (apiEnvConfig.isDefined) {
+      val envConfig = apiEnvConfig.get
+      if (envConfig.threadPoolSize < 1 || envConfig.threadPoolSize > 20) {
+        throw new BadConfigException("ThreadPoolSize should be a valid number between 0 to 20")
+      }
+      if (envConfig.apiWaitingTime < 60000) { // 60000ms = 1 mint
```

Comment:
should have a reasonable max -- say 15 mins? Does that seem right?

Reply:
done